lib/msal-common/src/app/module/AuthorizationCodeModule.ts

```diff
@@ -123,15 +123,23 @@ export class AuthorizationCodeModule extends AuthModule {
             this.cryptoObj
         );
 
-        const acquiredTokenResponse = this.networkClient.sendPostRequestAsync(
+        const acquiredTokenResponse = await this.networkClient.sendPostRequestAsync<ServerAuthorizationTokenResponse>(
             tokenEndpoint,
             {
                 body: tokenReqParams.createRequestBody(),
                 headers: tokenReqParams.createRequestHeaders()
             }
         );
 
-        return null;
+        try {
+            validateServerAuthorizationTokenResponse(acquiredTokenResponse);
+            const responseHandler = new ResponseHandler(this.clientConfig.auth.clientId, this.cacheStorage, this.cacheManager, this.cryptoObj);
+            return responseHandler.createTokenResponse(acquiredTokenResponse, tokenReqParams.state);
+        } catch (e) {
+            this.cacheManager.resetTempCacheItems(tokenReqParams.state);
```

Comment:
nitpick: can we merge all error unroll (account = null and cache reset) to a helper API if used at multiple places?

Reply:
this wouldn't change anything - resetTempCacheItems is the only thing that needs to be called. We can review at the end if this API would add anything, but as it is now it doesn't add any value.